pkg/api/v1/types.go

```diff
@@ -1649,6 +1649,8 @@ type Endpoints struct {
 type EndpointSubset struct {
 	// IP addresses which offer the related ports.
 	Addresses []EndpointAddress `json:"addresses,omitempty"`
+	// IP addresses which offer the related ports but have not yet passed readiness checks
+	UnreadyAddresses []EndpointAddress `json:"unreadyAddresses,omitempty"`
```

Comment:
Nice way to address backward compatibility without redundant info or a new virtual resource. I like it. We can consider refactoring the info in the v2 API -- we at least know we'll need to shard endpoints into multiple resources (in a stable fashion, such as by using subset and IP prefix).